backend/lcfs/web/api/transaction/repo.py

```diff
@@ -23,46 +29,69 @@ def __init__(self, db: AsyncSession = Depends(get_async_db_session)):
+        transfer_type_condition = TransactionView.transaction_type == 'Transfer'
 
-            # Include "Draft" transactions if organization_id matches from_organization_id
-            draft_condition = and_(
+        if organization_id is not None:
+            # Fetch visible statuses for both roles for "Transfer" transactions
+            visible_statuses_transferor = await self.get_visible_statuses(EntityType.Transferor)
+            visible_statuses_transferee = await self.get_visible_statuses(EntityType.Transferee)
+
+            # Construct role-specific conditions, applying them only to "Transfer" transactions
+            transferor_condition = and_(
+                transfer_type_condition,
                 TransactionView.from_organization_id == organization_id,
-                TransactionView.status == "Draft"
+                TransactionView.status.in_([status.value for status in visible_statuses_transferor])
             )
-            
-            # Combine the base condition with the draft inclusion condition
-            combined_condition = or_(base_condition, draft_condition)
-            conditions.append(combined_condition)
+            transferee_condition = and_(
+                transfer_type_condition,
+                TransactionView.to_organization_id == organization_id,
+                TransactionView.status.in_([status.value for status in visible_statuses_transferee])
+            )
+
+            # For transactions that are not of type "Transfer", include them without visibility filtering
+            non_transfer_condition = and_(
+                TransactionView.transaction_type != 'Transfer',
+                TransactionView.to_organization_id == organization_id
+            )
+
+            # Combine conditions since an organization can be both transferor and transferee, or neither for non-"Transfer" transactions
+            combined_role_condition = or_(transferor_condition, transferee_condition, non_transfer_condition)
+            query_conditions.append(combined_role_condition)
         else:
-            # If no organization_id is provided, condition to exclude transactions for gov users
-            conditions.append(TransactionView.status.not_in([
-                "Draft", 
-                "Sent", 
-                "Deleted",
-                "Declined"
-            ]))
+            # Fetch visible statuses for government, but only for "Transfer" transactions
+            visible_statuses_government = await self.get_visible_statuses(EntityType.Government)
+            government_condition = and_(
+                transfer_type_condition,
+                TransactionView.status.in_([status.value for status in visible_statuses_government])
+            )
+
+            # Include non-"Transfer" transactions for government without applying visibility filtering
+            gov_non_transfer_condition = TransactionView.transaction_type != 'Transfer'
```

Comment:
leave a TODO here because there will be additional visibility checks needed for other transaction types.